src/librbd/image/RemoveRequest.cc

```diff
@@ -621,6 +627,24 @@ void RemoveRequest<I>::handle_remove_v1_image(int r) {
   if (!m_old_format) {
     remove_v2_image();
   }
+
+}
+
+template<typename I>
+void RemoveRequest<I>::remove_image_cache() {
```

Comment:
The vast majority of the time, the remove request will be invoked from a node where the image has never been run. What really needs to happen is that the shared parent cache should be able to be garbage collected on a hypervisor host when needed (or when it eventually discovers that the image no longer exists).

Reply:
Yeah -- that would definitely work as well.